pkg/controller/servicednsendpoint/servicednsendpoint_controller_test.go

```diff
@@ -0,0 +1,279 @@
+				{DNSName: c2RegionDNSName, Targets: []string{globalDNSName}, RecordType: RecordTypeCNAME, RecordTTL: minDNSTTL},
+				{DNSName: c2ZoneDNSName, Targets: []string{c2RegionDNSName}, RecordType: RecordTypeCNAME, RecordTTL: minDNSTTL},
+			},
+			expectError: false,
+		},
+		"NoLBInBothClusters": {
+			dnsObject: feddnsv1a1.MultiClusterServiceDNSRecord{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+				},
+				Spec: feddnsv1a1.MultiClusterServiceDNSRecordSpec{
+					FederationName: federation,
+					DNSSuffix:      dnsZone,
+				},
+				Status: feddnsv1a1.MultiClusterServiceDNSRecordStatus{
+					DNS: []feddnsv1a1.ClusterDNS{
+						{
+							Cluster: c1, Zone: c1Zone, Region: c1Region,
+						},
+						{
+							Cluster: c2, Zone: c2Zone, Region: c2Region,
+						},
+					},
+				},
+			},
+			expectEndpoints: []*feddnsv1a1.Endpoint{
+				{DNSName: c1RegionDNSName, Targets: []string{globalDNSName}, RecordType: RecordTypeCNAME, RecordTTL: minDNSTTL},
+				{DNSName: c1ZoneDNSName, Targets: []string{c1RegionDNSName}, RecordType: RecordTypeCNAME, RecordTTL: minDNSTTL},
+				{DNSName: c2RegionDNSName, Targets: []string{globalDNSName}, RecordType: RecordTypeCNAME, RecordTTL: minDNSTTL},
+				{DNSName: c2ZoneDNSName, Targets: []string{c2RegionDNSName}, RecordType: RecordTypeCNAME, RecordTTL: minDNSTTL},
+			},
+			expectError: false,
+		},
+		"HostnameInLB": {
+			dnsObject: feddnsv1a1.MultiClusterServiceDNSRecord{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+				},
+				Spec: feddnsv1a1.MultiClusterServiceDNSRecordSpec{
+					FederationName: federation,
+					DNSSuffix:      dnsZone,
+				},
+				Status: feddnsv1a1.MultiClusterServiceDNSRecordStatus{
+					DNS: []feddnsv1a1.ClusterDNS{
+						{
+							Cluster: c1, Zone: c1Zone, Region: c1Region,
+							LoadBalancer: v1.LoadBalancerStatus{Ingress: []v1.LoadBalancerIngress{{Hostname: "a9.us-west-2.elb.amazonaws.com"}}},
+						},
+						{
+							Cluster: c2, Zone: c2Zone, Region: c2Region,
+							LoadBalancer: v1.LoadBalancerStatus{Ingress: []v1.LoadBalancerIngress{{IP: lb2}}},
+						},
+					},
+				},
+			},
+			expectEndpoints: []*feddnsv1a1.Endpoint{
+				{DNSName: globalDNSName, Targets: []string{lb2, lb3}, RecordType: RecordTypeA, RecordTTL: minDNSTTL},
+				{DNSName: c1RegionDNSName, Targets: []string{lb3}, RecordType: RecordTypeA, RecordTTL: minDNSTTL},
+				{DNSName: c1ZoneDNSName, Targets: []string{lb3}, RecordType: RecordTypeA, RecordTTL: minDNSTTL},
+				{DNSName: c2RegionDNSName, Targets: []string{lb2}, RecordType: RecordTypeA, RecordTTL: minDNSTTL},
+				{DNSName: c2ZoneDNSName, Targets: []string{lb2}, RecordType: RecordTypeA, RecordTTL: minDNSTTL},
+			},
+			expectError: false,
+		},
+	}
+
+	for testName, tc := range testCases {
+		t.Run(testName, func(t *testing.T) {
+			endpoints, err := GetEndpointsForServiceDNSObject(&tc.dnsObject)
```

Comment:
(No action required) I think this testing could be both simpler and more comprehensive if each non-trivial function were tested from the bottom up rather than only validating the behavior of `GetEndpointsForServiceDNSObject`.  Please consider applying this approach to future testing efforts.